crates/bevy_ecs/src/schedule_v3/state.rs

```diff
@@ -0,0 +1,50 @@
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::mem;
+
+use crate as bevy_ecs;
+use crate::change_detection::Mut;
+use crate::schedule_v3::{ScheduleLabel, SystemSet, WorldExt};
+use crate::system::Resource;
+use crate::world::World;
+
+/// Types that can define states in a finite-state machine.
+pub trait Statelike: 'static + Send + Sync + Clone + PartialEq + Eq + Hash + Debug {}
+impl<T> Statelike for T where T: 'static + Send + Sync + Clone + PartialEq + Eq + Hash + Debug {}
+
+/// TBD
```

Comment:
```suggestion
/// Systems with this [`ScheduleLabel`] will run when the [`CurrentState`] of `S` is the value in the tuple struct.
```

Reply:
Not confident on the exact mechanics here, but should help start you on docs here.